### ISSUE TITLE
Absence of seed for testing in sample in sympy.stats.rv

### ISSUE THREAD

**Issue by czgdp1807** [sympy/sympy#16741]
## References
https://travis-ci.org/sympy/sympy/jobs/517841957

## Brief Description
The above job is failing because the current tests pass wrong parameters to the `LogNormal` distribution as shown below,
```
# Test sampling: Only e^mean in sample std of 0
    for i in range(3):
        X = LogNormal('x', i, 0)
        assert S(sample(X)) == N(exp(i))

```
The standard deviation of zero allows same random number to be generated which is required for testing. However, according to [Wikipedia page](https://en.wikipedia.org/wiki/Log-normal_distribution) and [this comment](https://github.com/sympy/sympy/pull/16571#discussion_r273540554), standard deviation of Lognormal cannot be zero. However, passing non-zero scale will make testing of sample difficult.
I did some deubugging and found out the standard `random` of python is used for sampling. So, for testing with correct parameters a provision to pass seed in `sample` should be allowed. 
Another way, to make testing easy is to allow 0 scale, though, which is conceptually wrong.

**Comment by czgdp1807** [comment]
@Upabjojr @jksuom Please provide your comments and solutions to the problem I have mentioned above.

**Comment by jksuom** [comment]
I would try setting the seed immediately before the test (assuming that the sampling will succeed in the first place).

**Comment by czgdp1807** [comment]
Yeah, that can be a fix. However, there should be some provision for those who don't know which methods/functions is used for sampling in the codebase, like, passing seed through the interface methods(`sample`, etc.). It will require of work to see how the control flows through different modules.
